python/mlx/nn/layers/positional_encoding.py

```diff
@@ -79,10 +87,15 @@ def __call__(self, x, offset: int = 0):
 
     @staticmethod
     def create_cos_sin_theta(
-        N: int, D: int, offset: int = 0, base: float = 10000, dtype=mx.float32
+        N: int,
+        D: int,
+        offset: int = 0,
+        base: float = 10000,
+        scale: float = 1.0,
+        dtype=mx.float32,
     ):
         D = D // 2
-        positions = mx.arange(offset, N, dtype=dtype)
+        positions = mx.arange(offset, N, dtype=dtype) * scale
```

Comment:
When I've seen this parameter before it usually is the denominator in a division rather than multiplied with. So you would set `scale = 4.0` and then `positions = positions * (1 / scale)`.

What do you think? Do you have examples where it is used?

Reply:
I agree with you if the user is passing integers, I assumed that the user will pass the scale fraction so that I can multiply it directly with the positions without the need to do `(1 / scale)`. What do you think? Should I change it to integers and do `(1 / scale)` before multiplying or just assume that the user will pass `(1 / scale)` directly?

Reply:
I think we can keep it the way it is. Calling it a scale makes it sound like it directly multiplies the positions (rather than divides them), so from a naming perspective the way you have it makes sense.